packages/next/build/index.js

```diff
@@ -11,9 +12,14 @@ const writeFile = promisify(fs.writeFile)
 
 export default async function build (dir, conf = null) {
   const config = loadConfig(PHASE_PRODUCTION_BUILD, dir, conf)
-  const buildId = await config.generateBuildId() // defaults to a uuid
   const distDir = join(dir, config.distDir)
 
+  let buildId = await config.generateBuildId() // defaults to a uuid
+  if (buildId == null) {
```

Comment:
`==` is intentional here to catch `null` or `undefined` (either would mess up any potential string methods)

or would you prefer to make this more explicit like `typeof buildId == "string"`?